library/Booster/Syntax/ParsedKore/Parser.y

```diff
@@ -419,11 +403,6 @@ the result. Namely, \\and, \\or, \\implies, and \\iff. Designed to be passed to
 foldl1' or foldr1.
 -}
 mkApply :: Token -> [Sort] -> KorePattern -> KorePattern -> KorePattern
-mkApply tok [sort] first second
-    | Token _ TokenAnd <- tok = KJAnd sort [first, second]
-    | Token _ TokenOr <- tok = KJOr sort [first, second]
-    | Token _ TokenImplies <- tok = KJImplies sort first second
-    | Token _ TokenIff <- tok = KJIff sort first second
 mkApply tok sorts@[_, _] first second
     | Token _ (TokenIdent _) <- tok = KJApp (mkId tok) sorts [first, second]
 mkApply other _ _ _ = error $ "mkApply: unsupported token " <> show other
```

Comment:
(nit-pick) This could now be inlined into its only use site at `mkAssoc` (and `mkAssoc` likewise I guess)